### PR TITLE
SNOW-660004: Add notes to session.table() about table name.

### PR DESCRIPTION
**Pull request by sfc-gh-yixie**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-660004

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Add text in the API doc to tell users to double quote the table name if it has special chars.

### CODE REVIEW

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/session.py [hunk 1]
```diff
@@ -892,6 +892,11 @@ def table(self, name: Union[str, Iterable[str]]) -> Table:
             name: A string or list of strings that specify the table name or
                 fully-qualified object identifier (database name, schema name, and table name).
 
+            Note:
+                If your table name has special characters, please double quote them like ``session.table('"my table"')``.
+                For fully qualified names, you need to double quote them separately like ``session.table('"my db"."my schema"."my.table"')``.
+                Refer to `Idenfier Requirements <https://docs.snowflake.com/en/sql-reference/identifiers-syntax.html>`_.
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/session.py:898]: typo: Idenfier -> Identifier

File: src/snowflake/snowpark/session.py [hunk 2]
```diff
@@ -892,6 +892,11 @@ def table(self, name: Union[str, Iterable[str]]) -> Table:
             name: A string or list of strings that specify the table name or
                 fully-qualified object identifier (database name, schema name, and table name).
 
+            Note:
+                If your table name has special characters, please double quote them like ``session.table('"my table"')``.
```
Comment (sfc-gh-jdu) [inline src/snowflake/snowpark/session.py:896]:
```suggestion
                If your table name has special characters, please use double quotes to mark it like ``session.table('"my table"')``.
```
I'm not sure if "double quote" can be used as a verb?

File: src/snowflake/snowpark/session.py (hunk 1 above)
Comment (sfc-gh-yixie) [reply src/snowflake/snowpark/session.py]: Good catch. Updated

File: src/snowflake/snowpark/session.py (hunk 2 above)
Comment (sfc-gh-yixie) [reply src/snowflake/snowpark/session.py]: I think it makes sense. Anyways I changed to your version.

File: src/snowflake/snowpark/session.py [hunk 3]
```diff
@@ -892,6 +892,11 @@ def table(self, name: Union[str, Iterable[str]]) -> Table:
             name: A string or list of strings that specify the table name or
                 fully-qualified object identifier (database name, schema name, and table name).
 
+            Note:
+                If your table name has special characters, please use double quotes to mark it like ``session.table('"my table"')``.
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/session.py:896]: Optionally, I suggest changing "has" to "contains". 
Remove "please".
Add "this," after "mark it like".

**Review by sfc-gh-mabrennan — COMMENTED** [review]
I only had a couple of small suggestions about the docstrings. Thanks.

**Review by sfc-gh-jdu — APPROVED** [review]

**Pull request merged**